transcom/mymove
# MB-16357 populate SITDestinationFinalAddress when TOO approves DDDSIT

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16357)

## Summary
Work done in [this pr](https://github.com/transcom/mymove/pull/10884) should ensure that the shipment destination address is populated with the new duty location address when the destination address is unknown. When [this pr](https://github.com/transcom/mymove/pull/10866) is merged, it will backfill shipment destination addresses. In this PR, we now make a copy of the streets/city/state/zip/country of the shipment destination address and set that to be the SITDestinationOriginalAddress and SITDestinationFinalAddress when the TOO approves a `DDDSIT` service item that doesn't have a `SITDestinationFinalAddress` 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Login as the prime simulator and look at `REWAYD` (or any other move where the shipment destination address is known)
2. Create a SIT destination service item without a delivery address
3. Login as a TOO and look at `REWAYD`
4. In the network tab of devtools, you should see that the DDDSIT service item doesn't have `sitDestinationOriginalAddress` or `sitDestinationFinalAddress`
5. Approve the shipment
6. In devtools, you should see that the `DDDSIT` service item will have `sitDestinationOriginalAddress` and `sitDestinationFinalAddress` and the streets/city/state/postal/country values match the values on the shipment destination address
7. Verify that `sitDestinationOriginalAddress` and `sitDestinationFinalAddress` ID (they'll be the same ID at this point) is different from the ID on the shipment destination address
 

### Backend
